electionleaflets/apps/leaflets/models.py

```diff
@@ -92,24 +93,62 @@ def save(self, *args, **kwargs):
 
 
 class LeafletImage(models.Model):
+    ORIENTATION_CHOICES = (
+        (1, 'Horizontal (normal)'),
+        (2, 'Mirror horizontal'),
+        (3, 'Rotate 180'),
+        (4, 'Mirror vertical'),
+        (5, 'Mirror horizontal and rotate 270 CW'),
+        (6, 'Rotate 90 CW'),
+        (7, 'Mirror horizontal and rotate 90 CW'),
+        (8, 'Rotate 270 CW'),
+    )
+
     leaflet = models.ForeignKey(Leaflet, related_name='images')
     image = ImageField(upload_to="leaflets", max_length=255)
     raw_image = ImageField(upload_to="raw_leaflets", blank=True, max_length=255)
     legacy_image_key = models.CharField(max_length=255, blank=True)
     image_type = models.CharField(choices=constants.IMAGE_TYPES,
         null=True, blank=True, max_length=255)
     image_text = models.TextField(blank=True)
+    orientation = models.PositiveSmallIntegerField(choices=ORIENTATION_CHOICES, default=1)
+    exif_data = models.BinaryField(blank=True)
 
     class Meta:
         ordering = ['image_type']
 
+    def _remove_exif_data(self):
+        full_image = Image.open(self.image)
+
+        if full_image.info.get('exif'):
+            exif_dict = piexif.load(full_image.info['exif'])
+            orientation = exif_dict.get('0th', {}).get(piexif.ImageIFD.Orientation, 1)
+
+            self.orientation = orientation
+            self.exif_data = full_image.info['exif']
+        else:
+            self.exif_data = b''
+
+        data = list(full_image.getdata())
+        image_without_exif = Image.new(full_image.mode, full_image.size)
+        image_without_exif.putdata(data)
+
+        new_file = BytesIO()
+        image_without_exif.save(new_file, 'jpeg')
+        file_content = ContentFile(new_file.getvalue())
+
+        self.image.save(self.image.name, file_content, save=False)
+
     def save(self, *args, **kwargs):
         """
         Save a copy of the raw_image as soon as possible.
 
         This is so we don't destroy images, by cropping them too small
         for example.
         """
+        if not self.exif_data:
```

Comment:
If the uploaded image doesn't have exif data, will this get called every time the model is saved? Is that ok?